initializr-generator/src/main/java/io/spring/initializr/generator/ProjectGenerator.java

```diff
@@ -467,6 +467,11 @@ protected void generateGitIgnore(File dir, ProjectRequest request) {
 		// Java versions
 		model.put("java8OrLater", isJava8OrLater(request));
 
+		// Has JPA starter
+		if (request.getStyle().contains("data-jpa")) {
```

Comment:
The library can't have the notion of `data-jpa`. That's an identifer a service has chosen but we can't just refer to it in the library.

Reply:
Ok, got it. I didn't notice that those services are declared in a different module (`initializr-service`), so it makes sense that `initializr-generator` doesn't use that hardcoded `data-jpa`.

Reply:
> Right now, Initializr doesn't support adding a plugin if a particular language is being used and if another dependency has been added.

That's the key in the issue. There is a workaround of adding a `jpa` facet and look for that instead. Please give it a try but first I'd like to see some tests for those changes. Let me know if you need help. Thanks!